### PR TITLE
[Text Generation][V2] NonKVCachePipeline

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Description

Added the `TestGenerationPipelineNoKVCache`. This pipeline processes the prompt and returns the new token. That's it.
Its main functionality is mapping prompt tokens to logits, instrumental for computing the perplexity of the model given a dataset

### Testing
Updated the integration tests to cover the case of non-kv-cache inference.

### Example Use

```python
from deepsparse.v2.text_generation import TextGenerationPipelineNoCache

prompt = ["Some funny prompt", "Why are you so"]

pipeline = TextGenerationPipelineNoCache(model_path="hf:mgoin/TinyStories-1M-ds",
                                         onnx_model_name="model-orig.onnx",
                                         sequence_length=20)

out = pipeline(prompt=prompt,
               include_prompt_logits=True,
               generation_kwargs=dict(output_scores=True))

for gen in out.generations:
    print(gen)
```

```bash
text='.' score=array([[ 2.9344807 , -0.03345669, -4.11256   , ..., -6.9316325 ,
        -4.6005425 ,  1.1827914 ],
       [ 7.008805  , -0.11603884, -7.1837015 , ..., -7.0405912 ,
        -2.386351  , -2.2007818 ],
       [ 6.348213  , -2.2960157 , -6.433192  , ..., -6.5930486 ,
        -5.8315077 , -0.58804405],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ]], dtype=float32) finished=True finished_reason='length' # notice that logits get zero padding from the end, this is because all logits need to have the same shape (the length of the longest prompt in the input +1)
text=' sad' score=array([[ 2.560934 ,  1.1993233, -6.670935 , ..., -7.3002615, -3.823823 ,
         1.8125833],
       [-1.1050931, -2.4256568, -7.3015127, ..., -6.1500154, -4.074909 ,
         1.8155754],
       [ 6.172593 , -2.2252593, -9.146653 , ..., -7.70834  , -4.810748 ,
         0.3985293],
       [ 1.4988875,  1.0973434, -4.4714937, ..., -4.8026247, -1.1791464,
         1.6924176]], dtype=float32) finished=True finished_reason='length'
```

### Next steps
- Create a parent`TextGenerationPipeline` operator that can either choose to use the kv-cache or non-kv cache version of the pipeline, depending on the topology of the ONNX model
- Move the overwriting of the transformer inputs to some high-level function
- Use the V2 pipeline for Perplexity calculation
- swap GraphRouter for LinearRouter in `TextGenerationPipelineNoKVCache`

### CODE REVIEW

File: src/deepsparse/v2/text_generation/nl_engine_operator.py [hunk 1]
```diff
@@ -29,7 +31,12 @@
 )
 
 
-__all__ = ["NLEngineOperator", "NlEngineInput"]
+__all__ = [
+    "NlEngineOperator",
+    "NlEngineOperatorNoCache",
```
Comment (dsikka) [inline src/deepsparse/v2/text_generation/nl_engine_operator.py:36]: this should be its own operator in a separate file

**Review by dsikka — CHANGES_REQUESTED** [review]

File: src/deepsparse/v2/text_generation/nl_engine_operator.py (hunk 1 above)
Comment (dbogunowicz) [reply src/deepsparse/v2/text_generation/nl_engine_operator.py:36]: Why can't I create variations of the same operator in the same script? I mean it makes sense for them to "live close" to each other. This question is about both NlEngineOperator(NoCache) as well as TextGenerationPipeline(NoCache)

Comment (dbogunowicz) [reply src/deepsparse/v2/text_generation/nl_engine_operator.py:36]: Come on

Comment (dsikka) [reply src/deepsparse/v2/text_generation/nl_engine_operator.py:36]: ⛔

Comment (dbogunowicz) [reply src/deepsparse/v2/text_generation/nl_engine_operator.py:36]: ⚔️

File: src/deepsparse/v2/text_generation/pipeline.py [hunk 2]
```diff
@@ -39,6 +41,79 @@
...
+                sequence_length=sequence_length,
+                prompt_sequence_length=1,
+                token_generator=token_generator,
+            ),
+            GenerateNewTokenOperator(tokenizer=self.tokenizer, force_max_tokens=True),
+            CompileGenerations(),
+            JoinOutput(tokenizer=self.tokenizer),
+            ProcessOutputs(tokenizer=self.tokenizer),
+        ]
+        router = LinearRouter(end_route=len(ops))
```
[the remark this answers is not in the document]
Comment (dsikka) [reply src/deepsparse/v2/text_generation/pipeline.py]: If you had issues with the LinearRouter updates, could you post here what they were?

[the remark this answers is not in the document]
Comment (dbogunowicz) [reply src/deepsparse/v2/text_generation/pipeline.py]: Let's take this discussion here: https://github.com/neuralmagic/deepsparse/pull/1434/files

File: src/deepsparse/utils/onnx.py [hunk 3]
```diff
@@ -564,7 +549,7 @@ def overwrite_onnx_model_inputs_for_kv_cache_models(
         else:
             raise ValueError(f"Unexpected external input name: {external_input.name}")
 
-    _LOGGER.debug(
+    _LOGGER.info(
```
Comment (dbogunowicz) [inline src/deepsparse/utils/onnx.py:552]:
```suggestion
    _LOGGER.debug(
```
